eng/common/scripts/Verify-AgentOS.ps1

```diff
@@ -0,0 +1,34 @@
+param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentPool,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentImage
+)
+
+$lookup = @(
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "windows-2019"; AgentOSMatch = $IsWindows }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "vs2017-win2016"; AgentOSMatch = $IsWindows }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "ubuntu-latest"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "ubuntu-20.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "ubuntu-18.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "ubuntu-16.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "macOS-latest"; AgentOSMatch = $IsMacOs }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "macOS-10.05"; AgentOSMatch = $IsMacOs }
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "macOS-10.04"; AgentOSMatch = $IsMacOs }
+    @{ AgentPool = "azsdk-pool-mms-ubuntu-1604-general"; AgentImage = "MMSUbuntu16.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "azsdk-pool-mms-ubuntu-1804-general"; AgentImage = "MMSUbuntu18.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "azsdk-pool-mms-ubuntu-2004-general"; AgentImage = "MMSUbuntu20.04"; AgentOSMatch = $IsLinux }
+    @{ AgentPool = "azsdk-pool-mms-win-2019-general"; AgentImage = "MMS2019"; AgentOSMatch = $IsWindows }
+)
+
+foreach ($mapping in $lookup)
```

Comment:
Should we also consider looking at the `Agent.Os` variable that devops set to make sure things as expected? Especially given we are starting to use that in conditions for various things.

Reply:
The ```Agent.OS``` parameter is actually just pulled from the local environment anyway

Where the variable is set: https://github.com/microsoft/azure-pipelines-agent/blob/763575e4443b61e2957d60dca9a9e068d29ccd92/src/Agent.Worker/JobRunner.cs#L132

Implementation detail: 
https://github.com/microsoft/azure-pipelines-agent/blob/763575e4443b61e2957d60dca9a9e068d29ccd92/src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs#L30
https://github.com/microsoft/azure-pipelines-agent/blob/f41d99d738767c46a2a907e5fce142e23b1c7602/src/Agent.Sdk/Util/PlatformUtil.cs#L26